### DISCUSSION TITLE
Table toggle cannot stop, always send request

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6331] [Help] [answered]
[table_toggle.webm](https://user-images.githubusercontent.com/28250303/234760653-d75fe448-da0a-4534-b83b-fa9ce26d2666.webm)

Steps:

1. Create any filament resource
2. add toggle column in table like below
3. Toggle 2 or 3 data **very fast**

```
Tables\Columns\ToggleColumn::make('is_active')
    ->label(__('message.active'))
    ->alignCenter()
    ->onIcon('heroicon-o-check')
    ->offIcon('heroicon-o-x')
    ->onColor('success')
    ->offColor('primary'),
```

How to prevent this?

**Comment by danharrin** [comment]
Maybe fixed by #6221?

**Comment by archilex** [comment]
Yes, this behavior should be fixed by #6221. Try copying and pasting the code to see if it helps. You can also try using the [ToggleIconColumn plugin](https://filamentphp.com/plugins/toggle-icon-column) that I’ve updated to use the fix in #6221. Try that and see if it fixes your issues. It’d be good to get some feedback on the fix.

**Comment by valpuia** [comment, marked as answer]
Yes, I think this #6221 fixed my problem.. I publish vendor and change `toggle-column` page like this [https://github.com/filamentphp/filament/blob/3d5df317e04bb786de8c169de5590e84dc64f09c/packages/tables/resources/views/columns/toggle-column.blade.php](https://github.com/filamentphp/filament/blob/3d5df317e04bb786de8c169de5590e84dc64f09c/packages/tables/resources/views/columns/toggle-column.blade.php) and this problem gone for me

Thank you @danharrin @archilex

**Reply by archilex** [reply]
Nice! Thanks for the confirmation! Just remember to delete your published view after the PR is merged to avoid potentially issues in the future as Filament gets updated.

**Reply by valpuia** [reply]
One thing I notice is, in dark mode, it doesn't use my `offColor` and replace with black..

**Reply by archilex** [reply]
I assume you're referring to Filament's toggleColumn? (Not my ToggleIconColumn)

**Reply by archilex** [reply]
Looking at the source, what you are seeing is default behavior. When you are in darkmode it defaults to `dark:bg-white/10` which is basically black.

**Reply by valpuia** [reply]
Yes Filament, for me if we set `color` it should be visible on both dark and light mode.. otherwise it doesn't feel great

But for now, completed and working.. thanks again for you work
